services/ec2/src/main/java/org/finra/gatekeeper/services/accessrequest/AccessRequestService.java

```diff
@@ -356,6 +427,163 @@ private void updateRequestDetails(Long requestId, String approverComments, Strin
+                        }
+                    }
+                }
+            } else {
+                logger.warn("Could not get request details for AccessRequest with ID: " + requestId + ". This request will not be returned ");
+            }
+        }
+
+        for (String k : userMap.keySet()) {
+            results.add(userMap.get(k));
+        }
+        return results;
+    }
+
+
+    private List<ActiveRequestUser> addExpiredRequest(List<ActiveRequestUser> activeRequestUserList, AccessRequest expiredRequest) {
+        if (expiredRequest != null) {
+            List<ActiveAccessRequest> linuxRequests = new ArrayList<>();
+            List<ActiveAccessRequest> windowsRequests = new ArrayList<>();
+
+            for(AWSInstance awsInstance : expiredRequest.getInstances()) {
+                if(awsInstance.getPlatform().equals("Linux")) {
+                    linuxRequests.add(new ActiveAccessRequest(expiredRequest.getId().toString(), awsInstance.getName(), awsInstance.getIp()));
+                }
+                if(awsInstance.getPlatform().equals("Windows")) {
+                    windowsRequests.add(new ActiveAccessRequest(expiredRequest.getId().toString(), awsInstance.getName(), awsInstance.getIp()));
+                }
+            }
+            ActiveAccessConsolidated expiredRequestsConsolidated = new ActiveAccessConsolidated(linuxRequests, windowsRequests);
+
+            for(ActiveRequestUser activeRequestUser : activeRequestUserList) {
+                ActiveAccessConsolidated activeRequestsConsolidated = activeRequestUser.getActiveAccess();
+                activeRequestUser.setExpiredAccess(expiredRequestsConsolidated);
+
+                for(AWSInstance awsInstance : expiredRequest.getInstances()) {
+                    List<ActiveAccessRequest> requests = new ArrayList<>();
+                    ActiveAccessRequest expiredAccessRequest = new ActiveAccessRequest(expiredRequest.getId().toString(), awsInstance.getName(), awsInstance.getIp());
+
+                    if(awsInstance.getPlatform().equals("Linux")) {
+                        requests = activeRequestUser.getActiveAccess().getLinux();
+                    }
+                    else if(awsInstance.getPlatform().equals("Windows")) {
+                        requests = activeRequestUser.getActiveAccess().getWindows();
+                    }
+
+                    Iterator<ActiveAccessRequest> requestIterator = requests.iterator();
+                    while(requestIterator.hasNext()) {
+                        ActiveAccessRequest activeAccessRequest = requestIterator.next();
+                        if(activeAccessRequest.equals(expiredAccessRequest)) {
+                            logger.info("Removing expired access request: " + expiredAccessRequest);
+                            requestIterator.remove();
+                        }
+                    }
+
+                    if(awsInstance.getPlatform().equals("Linux")) {
```

Comment:
Can the code in this if/else block be merged into the code in the above if/else block?